### DISCUSSION TITLE
Multiple users access multiple data

### DISCUSSION THREAD

**Discussion by haoxl3** [Kitware/trame#340] [Q&A]
I want to make trame embedded in another web service written in javascript via iframe, when multiple users access trame there is a mutual impact issue. For example, user A first accesses the x.vtk file, user B then accesses the y.vtk file, and user A's screen also becomes the y.vtk file. I must start two trame services so that they will not be affected by each other. How can different users freely access different vtk files rendered by trame?
I found cookiecutter, which I didn't try successfully, and I didn't quite understand if it would solve my problem.

**Comment by jourdain** [comment]
You need a multi-user setup, which will start 1 process per user. This can be achieved by bundling your trame app within docker. The cookie-cutter provide you with the basic docker setup inside `./bundle/docker/*`. Otherwise you can also look at [that simple example](https://github.com/Kitware/trame/tree/master/examples/deploy/docker/SingleFile).
